src/python/foglamp/core/api.py

```diff
@@ -6,21 +6,117 @@
 
 import time
 from aiohttp import web
+from foglamp import configuration_manager
 
-__author__ = "Amarendra K. Sinha"
+__author__ = "Amarendra K. Sinha, Ashish Jabble"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
-__start_time = time.time()
+START_TIME = time.time()
+
+HELP = """
+    -----------------------------------------------------------------------
+    | GET             | /ping                                             |
+
+    | GET             | /categories                                       |
+    | GET             | /category/{category_name}                         |
+    | GET             | /category/{category_name}/{config_item}           |
+    | PUT DELETE      | /category/{category_name}/{config_item}/{value}   |
+
+    -----------------------------------------------------------------------
+"""
 
 
 async def ping(request):
     """
+
+    :param request:
     :return: basic health information json payload
     {'uptime': 32892} Time in seconds since FogLAMP started
     """
 
-    since_started = time.time() - __start_time
+    since_started = time.time() - START_TIME
 
     return web.json_response({'uptime': since_started})
+
+
+#################################
+###  Configuration Manager
+#################################
+
+async def get_categories(request):
+    """
+
+    :param request:
+    :return: the list of known categories in the configuration database
+    """
+    categories = await configuration_manager.get_all_category_names()
+    categories_json = [{"key": i[0], "description": i[1]} for i in categories]
+    return web.json_response({'categories': categories_json})
+
+
+async def get_category(request):
+    """
+
+    :param request:  category_name is required
+    :return: the configuration items in the given category.
+    """
+    category_name = request.match_info.get('category_name', None)
+    category = await configuration_manager.get_category_all_items(category_name)
+    # TODO: If category is None from configuration manager. Should we send category
```

Comment:
Good point, error handling is not covered in the document. I should have defined the error mechanism. My thought for now would be to say we return None for the first pass and add better error handling.